pkg/apis/pubsub/v1alpha1/pullsubscription_conversion_test.go

```diff
@@ -0,0 +1,222 @@
+)
+
+// These variables are used to create a 'complete' version of PullSubscription where every field is
+// filled in.
+var (
+	trueVal  = true
+	seconds  = int64(314)
+	duration = "30s"
+
+	completeObjectMeta = metav1.ObjectMeta{
+		Name:            "name",
+		GenerateName:    "generateName",
+		Namespace:       "namespace",
+		SelfLink:        "selfLink",
+		UID:             "uid",
+		ResourceVersion: "resourceVersion",
+		Generation:      2012,
+		CreationTimestamp: metav1.Time{
+			Time: time.Unix(1, 1),
+		},
+		DeletionTimestamp: &metav1.Time{
+			Time: time.Unix(2, 3),
+		},
+		DeletionGracePeriodSeconds: &seconds,
+		Labels:                     map[string]string{"steel": "heart"},
+		Annotations:                map[string]string{"New": "Cago"},
+		OwnerReferences: []metav1.OwnerReference{
+			{
+				APIVersion:         "apiVersion",
+				Kind:               "kind",
+				Name:               "n",
+				UID:                "uid",
+				Controller:         &trueVal,
+				BlockOwnerDeletion: &trueVal,
+			},
+		},
+		Finalizers:  []string{"finalizer-1", "finalizer-2"},
+		ClusterName: "clusterName",
+	}
+
+	completeURL = apis.URL{
+		Scheme:     "scheme",
+		Opaque:     "opaque",
+		User:       url.User("user"),
+		Host:       "host",
+		Path:       "path",
+		RawPath:    "rawPath",
+		ForceQuery: false,
+		RawQuery:   "rawQuery",
+		Fragment:   "fragment",
+	}
+
+	completeDestination = duckv1.Destination{
+		Ref: &duckv1.KReference{
+			APIVersion: "apiVersion",
+			Kind:       "kind",
+			Namespace:  "namespace",
+			Name:       "name",
+		},
+		URI: &completeURL,
+	}
+
+	completeSourceSpec = duckv1.SourceSpec{
+		Sink: completeDestination,
+		CloudEventOverrides: &duckv1.CloudEventOverrides{
+			Extensions: map[string]string{"supers": "reckoners"},
+		},
+	}
+
+	completeIdentitySpec = duckv1alpha1.IdentitySpec{
+		GoogleServiceAccount: "googleServiceAccount",
+	}
+
+	completeSecret = &v1.SecretKeySelector{
+		LocalObjectReference: v1.LocalObjectReference{
+			Name: "name",
+		},
+		Key:      "key",
+		Optional: &trueVal,
+	}
+
+	completePubSubSpec = duckv1alpha1.PubSubSpec{
+		SourceSpec:   completeSourceSpec,
+		IdentitySpec: completeIdentitySpec,
+		Secret:       completeSecret,
+		Project:      "project",
+	}
+
+	completeIdentityStatus = duckv1alpha1.IdentityStatus{
+		Status: duckv1.Status{
+			ObservedGeneration: 7,
+			Conditions: duckv1.Conditions{
+				{
+					Type:   "Ready",
+					Status: "True",
+				},
+			},
+		},
+		ServiceAccountName: "serviceAccountName",
+	}
+
+	completePubSubStatus = duckv1alpha1.PubSubStatus{
+		IdentityStatus: completeIdentityStatus,
+		SinkURI:        &completeURL,
+		CloudEventAttributes: []duckv1.CloudEventAttributes{
+			{
+				Type:   "type",
+				Source: "source",
+			},
+		},
+		ProjectID:      "projectID",
+		TopicID:        "topicID",
+		SubscriptionID: "subscriptionID",
+	}
+
+	// completePullSubscription is a PullSubscription with every filled in, except TypeMeta.
```

Comment:
nit: `every field filled in`

Reply:
https://github.com/google/knative-gcp/pull/813